### PR TITLE
klavaro: 3.08 -> 3.10 

### PR DESCRIPTION
**Pull request by davidak**
###### Motivation for this change

Update and fixes

Screenshot from 2020-05-28 00 44 18

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc maintainer @mimame

### CODE REVIEW

File: pkgs/games/klavaro/default.nix [hunk 1]
```diff
@@ -17,14 +17,19 @@ stdenv.mkDerivation rec {
...
-  meta = {
+  meta = with stdenv.lib; {
     description = "Just another free touch typing tutor program";
     homepage = "http://klavaro.sourceforge.net/";
-    license = stdenv.lib.licenses.gpl3Plus;
-    platforms = stdenv.lib.platforms.linux;
-    maintainers = [stdenv.lib.maintainers.mimame];
+    license = licenses.gpl3Plus;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ mimame ];
```
Comment (drewrisinger) [inline pkgs/games/klavaro/default.nix:33]: Would you consider adding yourself as a maintainer?

File: pkgs/games/klavaro/default.nix [hunk 2]
```diff
@@ -1,12 +1,12 @@
-{ stdenv, fetchurl, makeWrapper, pkgconfig, intltool, curl, gtk3 }:
+{ stdenv, fetchurl, makeWrapper, pkgconfig, intltool, curl, file, gtk3 }:
 
 stdenv.mkDerivation rec {
   pname = "klavaro";
-  version = "3.08";
+  version = "3.10";
```
Comment (drewrisinger) [inline pkgs/games/klavaro/default.nix:5]: Squash commits for 3.09 & 3.10

File: pkgs/games/klavaro/default.nix [hunk 3]
```diff
@@ -1,12 +1,12 @@
-{ stdenv, fetchurl, makeWrapper, pkgconfig, intltool, curl, gtk3 }:
+{ stdenv, fetchurl, makeWrapper, pkgconfig, intltool, curl, file, gtk3 }:
```
Comment (drewrisinger) [inline pkgs/games/klavaro/default.nix:1]: Formatting: best practice is to have each on a separate line, and roughly alphabetize everything after makeWrapper.

i.e.:
```
{ stdenv
, fetchurl
, makeWrapper
, curl
...
}:

**Review by drewrisinger — CHANGES_REQUESTED** [review]
- [x] Squash 3.09 & 3.10 commits (could honestly squash all these commits together)
- [x] Rename PR to something like ``klavaro: 3.08 -> 3.10``
- [ ] Haven't tested build locally

File: pkgs/games/klavaro/default.nix (hunk 1 above)
Comment (davidak) [reply pkgs/games/klavaro/default.nix]: ok, why not

**Comment by davidak** [comment]
@drewrisinger you can test with:

`nix run -f https://api.github.com/repos/davidak/nixpkgs/tarball/klavaro10 klavaro`

**Comment by drewrisinger** [comment]
Had the following issue on Ubuntu 18.04 (x86-64) using ``nix-review pr 89352``, but had similar issues with GUI apps before. Would appreciate an independent tester on NixOS @mimame 

```bash
$ nix-shell /home/artiqdev/.cache/nixpkgs-review/pr-89352/shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-89352]$ klavaro

(process:8519): Gtk-WARNING **: 10:18:52.985: Locale not supported by C library.
        Using the fallback 'C' locale.
Gtk-Message: 10:18:53.009: Failed to load module "canberra-gtk-module"
Gtk-Message: 10:18:53.010: Failed to load module "canberra-gtk-module"

(klavaro:8519): Gtk-WARNING **: 10:18:53.295: Could not load image 'other.png': Unable to load image-loading module: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: cannot open shared object file: No such file or directory

(klavaro:8519): Gtk-WARNING **: 10:18:53.345: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.

(klavaro:8519): Gtk-WARNING **: 10:18:53.364: Could not load image 'key.png': Unable to load image-loading module: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: cannot open shared object file: No such file or directory

(klavaro:8519): Gtk-WARNING **: 10:18:53.365: Could not load image 'altcolor.png': Unable to load image-loading module: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: cannot open shared object file: No such file or directory

(klavaro:8519): Gtk-WARNING **: 10:18:53.366: Could not load image 'progress.png': Unable to load image-loading module: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: cannot open shared object file: No such file or directory

(klavaro:8519): Gtk-WARNING **: 10:18:53.366: Could not load image 'top10.png': Unable to load image-loading module: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: cannot open shared object file: No such file or directory
**
Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/gnome/16x16/status/image-missing.png: Unable to load image-loading module: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: cannot open shared object file: No such file or directory (gdk-pixbuf-error-quark, 5)
Bail out! Gtk:ERROR:../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/gnome/16x16/status/image-missing.png: Unable to load image-loading module: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: /snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: cannot open shared object file: No such file or directory (gdk-pixbuf-error-quark, 5)
Aborted (core dumped)
```

**Comment by davidak** [comment]
@drewrisinger we can get it working on other systems later. the program should also work on windows and mac

any idea why it tries to open `/snap/code/31/usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so`?

**Comment by drewrisinger** [comment]
``nixpkgs-review`` must not be creating a pure shell, I must have some dependencies installed as Snaps.

I don't particularly care about this application, mostly just trying to review & confirm that the GUI app is working before approving.

File: pkgs/games/klavaro/default.nix (hunk 3 above)
Comment (Ma27) [reply pkgs/games/klavaro/default.nix]: Just for the record, I don't think we should ask contributors to perform those kind of nitpicks. We have tooling which automates that (e.g. `nixpkgs-fmt`) and I really don't think that a human should care about that kind of details during a review.

**Pull request merged**